composer/core/state.py

```diff
@@ -89,15 +98,14 @@ def __init__(
 
             # data configurations
             grad_accum: int,
-            train_dataloader: types.DataLoader,
-            eval_dataloader: types.DataLoader,
+            train_dataloader: Union[types.DataLoader, types.DataSpec],
```

Comment:
`State` should only keep the pytorch dataloader; the rest of the details can be in the trainer itself?

Reply:
Discussed offline that it'll remain attached to state -- algorithms might need the batch split function, for example.